pkg/cri/containerd/fake/runtime.go

```diff
@@ -0,0 +1,127 @@
+package fake
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cio"
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/oci"
+	"github.com/gogo/protobuf/types"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+type Client struct {
+	cache map[string]containerd.Container
+}
+
+type ClientOpt func(client *Client)
+
+func NewFakeClient(opts ...ClientOpt) *Client {
+	client := &Client{cache: map[string]containerd.Container{}}
+	for _, opt := range opts {
+		opt(client)
+	}
+	return client
+}
+
+func WithCachedContainer(containerID string, container containerd.Container) ClientOpt {
+	return func(client *Client) {
+		client.cache[containerID] = container
+	}
+}
+
+func (c Client) LoadContainer(_ context.Context, id string) (containerd.Container, error) {
+	container, wasFound := c.cache[id]
+	if wasFound {
+		return container, nil
+	}
+	return nil, fmt.Errorf("container not found: %s", id)
+}
+
+type Container struct {
+	id               string
+	missingLinuxInfo bool
+	netnsPath        string
+}
+
+func NewFakeContainer(id string, netnsPath string) *Container {
+	return &Container{
+		id:        id,
+		netnsPath: netnsPath,
+	}
+}
+
+func NewFakeContainerWithoutNetworkNamespace(id string) *Container {
+	return &Container{
+		id: id,
+	}
+}
+
+func NewFakeNonLinuxContainer(id string) *Container {
+	return &Container{
+		id:               id,
+		missingLinuxInfo: true,
+	}
+}
+
+func (c Container) ID() string {
+	return c.id
+}
+
+func (Container) Info(context.Context, ...containerd.InfoOpts) (containers.Container, error) {
+	return containers.Container{}, nil
+}
+
+func (Container) Delete(context.Context, ...containerd.DeleteOpts) error {
+	return nil
+}
+
+func (Container) NewTask(context.Context, cio.Creator, ...containerd.NewTaskOpts) (containerd.Task, error) {
+	return nil, nil
+}
```

Comment:
These are not used.

Reply:
But we are not passing this at that interface anywhere right ?

Reply:
Yes, but if my dummy does not implement all the methods on an interface ... It does not implement it. 

Maybe I don't understand the implications behind your question.